src/crypto/CryptoTests.cpp

```diff
@@ -248,63 +262,71 @@ TEST_CASE("StrKey tests", "[crypto]")
     size_t n_corrupted = 0;
     size_t n_detected = 0;
 
-    for (int round = 0; round < 5; round++)
+    for (int round = 0; round < 100; round++)
     {
         const int expectedSize = 32;
         std::vector<uint8_t> in(input(expectedSize));
         std::string encoded = strKey::toStrKey(version, in).value;
 
         for (size_t p = 0u; p < encoded.size(); p++)
         {
+            // perform a single corruption
             for (int st = 0; st < 4; st++)
             {
                 std::string corrupted(encoded);
                 auto pos = corrupted.begin() + p;
                 switch (st)
                 {
                 case 0:
-                    if (corrupted[p] == 'A' && p + 1 == encoded.size())
-                    {
-                        // trailing 'A' is equivalent to 0 (and can be dropped)
-                        continue;
-                    }
-                    else
-                    {
-                        corrupted.erase(pos);
-                        break;
-                    }
+                    // remove
+                    corrupted.erase(pos);
+                    break;
                 case 1:
-                    corrupted[p]++;
+                    // modify
+                    corrupted[p] = randomB32();
                     break;
                 case 2:
+                    // duplicate element
                     corrupted.insert(pos, corrupted[p]);
                     break;
-                default:
-                    if (p > 0 && corrupted[p] != corrupted[p - 1])
+                case 3:
+                    // swap consecutive elements
+                    if (p > 0)
                     {
                         std::swap(corrupted[p], corrupted[p - 1]);
                     }
-                    else
-                    {
-                        continue;
-                    }
+                    break;
+                default:
+                    abort();
                 }
                 uint8_t ver;
                 std::vector<uint8_t> dt;
                 if (corrupted != encoded)
                 {
-                    n_corrupted++;
+                    bool sameSize = (corrupted.size() == encoded.size());
+                    if (sameSize)
```

Comment:
Why is this check needed now when it wasn't needed before? Same for line 314.

Reply:
because now the test computes the stats for crc related failures (same size, different content), and enforces (REQUIRE) the other types of corruption